sql/core/src/main/scala/org/apache/spark/sql/ForeachWriter.scala

```diff
@@ -50,14 +50,13 @@ import org.apache.spark.annotation.Evolving
  *
  * Important points to note:
  * <ul>
- * <li>The `partitionId` and `epochId` can be used to deduplicate generated data when failures
- *     cause reprocessing of some input data. This depends on the execution mode of the query. If
- *     the streaming query is being executed in the micro-batch mode, then every partition
- *     represented by a unique tuple (partitionId, epochId) is guaranteed to have the same data.
- *     Hence, (partitionId, epochId) can be used to deduplicate and/or transactionally commit data
- *     and achieve exactly-once guarantees. However, if the streaming query is being executed in the
- *     continuous mode, then this guarantee does not hold and therefore should not be used for
- *     deduplication.
+ * <li>Spark doesn't guarantee same output for (partitionId, epochId) on failure, so deduplication
+ *     cannot be achieved with (partitionId, epochId). Refer SPARK-28650 for more details.
```

Comment:
SPARK-28650 has only the following content, could we remove this `Refer SPARK-28650 for more details` by embedding this information?
> But we can break this easily actually when restarting a query but a batch is re-run (e.g., upgrade Spark)
Source returns a different DataFrame that has a different partition number (e.g., we start to not create empty partitions in Kafka Source V2).
A new added optimization rule may change the number of partitions in the new run.
Change the file split size in the new run.
Since we cannot guarantee that the same (partitionId, epochId) has the same data. We should update the document for "ForeachWriter".

Reply:
I'd inline a few of those examples of when it can occur, and leave it at that.

Reply:
While we would want to warn for existing ForeachWriter users (this can't be done in javadoc - it should be noticed via release note, streaming doc guide - maybe?, etc), not all end users would like to know the details. (I'd expect there would be more users ignoring the details than trying to understand all the details.) As they could still get the details from visiting the issue SPARK-28650, I'd borrow some example as `e.g.` from there and leave reference.